Tutorials/Creating-a-Custom-Dashboard/index.md

```diff
@@ -0,0 +1,333 @@
+
+We inject the logResource into our controller:
+
+    angular.module("umbraco").controller("CustomWelcomeDashboardController", function ($scope, userService, logResource) {
+
+Add a property on our ViewModel to store the log information:
+
+    vm.LogEntries = [];
+
+Add to our WelcomeDashboard.html view some markup using angular's *ng-repeat* to display a list of these log entries:
+
+    <h2>We know what you edited last week...</h2>
+        <ul>
+            <li ng-repeat="logEntry in vm.LogEntries">{{logEntry.nodeId}} - {{logEntry.comment}} - {{logEntry.timestamp  | date:'medium'}}</li>
+        </ul>
+
+Back in our controller we'll populate the array of entries using the **logResource**, returning any log entries of 'Save' type:
+
+     logResource.getUserLog("save",new Date())
+       .then(function (response) {
+            console.log(response)
+            vm.LogEntries = response;
+       });
+
+However looking at the console output will reveal the data retrieved by the **logResource** is a little sparse eg:
+
+    Object
+        $$hashKey:"265"
+        Content:
+            Object
+                comment:"Save Content performed by user"
+                logType:1
+                nodeId:1063
+                timestamp:"2017-03-18T14:09:40.91"
+                userId:0
+
+There is a bit of work to be done to provide something meaningful to the editor from the audit log!
+
+We can use the **entityResource**, an Umbraco Angular resource that enables us to retrieve more information about an entity given it's id.
+
+Inject this into our angular controller:
+
+    angular.module("umbraco").controller("CustomWelcomeDashboardController", function ($scope, userService, logResource, entityResource) {
+
+We need to loop through the response from the **logResource**, filter out 'saves' we're not interested in eg, Macro Saves, or DocType Saves, generally we need the entry in the log to have a nodeId and mention either Media or Content in the comment text. 
+
+The **entityResource** then has a **getById** method that accepts the Id of the item and the entity 'type' to retrieve useful information about the entity, ie it's name and icon.
+
+Putting this together:
+
+       logResource.getUserLog("save", new Date()).then(function (response) {
+            console.log(response);
+            var logEntries = [];
+            // loop through the response, and filter out save log entries we are not interested in
+            angular.forEach(response, function (item) {
+                // if no entity exists -1 is returned for the nodeId (eg saving a macro would create a log entry without a nodeid)
+                if (item.nodeId > 0) {
+                    //this is the only way to tell them apart!!
+                    if (item.comment.indexOf("Media") > -1 || item.comment.indexOf("Content") > -1) {
```

Comment:
ugh, yes, had forgotten about that, and also not thought it through - it would be good if it said the type of thing that it was in the logs :-P - so would it be better than a space - to use a regex match ?

 if (item.comment.match("(\\bContent\\b|\\bMedia\\b)")) {
//do stuff for entries that mention Content or Media but not ContentType/MediaType....

what do you think?